.drone.yml

```diff
@@ -1297,6 +1297,99 @@ volumes:
+
+environment:
+  TEST_PROTO_DIR_SUFFIX: _012
+  PYTHON_LOGS_SERVER_ADDRESS: 65.21.119.66
+  PYTHON_LOGS_SERVER_PATH: /sftp-data/tezedgeci/upload/tezedge
+
+defs:
+  - &python-test
+    image: tezedge/tezedge-ci-builder:rust-1.58.1-v13.0-octez
+    pull: if-not-exists
+    user: root
+    environment:
+      SODIUM_USE_PKG_CONFIG: 1
+      RUST_BACKTRACE: 1
+      RUN_TEZEDGE_BAKER: "external"
+    commands:
+      - cd tezos-src/tezos/tests_python
+      - log_dir=/drone/src/python-test-logs/$${DRONE_STEP_NAME}_logs
+      - mkdir -p $log_dir && echo "Logs are place on $${PYTHON_LOGS_SERVER_ADDRESS} - $${PYTHON_LOGS_SERVER_PATH}}"
+      - poetry run pytest -s --log-dir=$log_dir tests$${TEST_PROTO_DIR_SUFFIX}/$${PYTHON_TEST}
+
+steps:
+
+  - name: prepare-data
+    image: alpine/git
+    user: root
+    commands:
+      - mkdir -p drone-cache
+
+  - name: restore-cache
+    image: meltwater/drone-cache
+    pull: true
+    environment:
+      SFTP_USERNAME:
+        from_secret: sftp_username
+      SFTP_PASSWORD:
+        from_secret: sftp_password
+      SFTP_AUTH_METHOD: PASSWORD
+      SFTP_HOST: 65.21.119.66
+      SFTP_PORT: 22
+      SFTP_CACHE_ROOT: "/upload"
+    settings:
+      restore: true
+      debug: true
+      backend: "sftp"
+      cache_key: 'build-{{ .Build.Number }}'
+      mount:
+        - 'drone-cache' # <- builds a cache from this directory
+
+  - name: prepare-python-tests
+    image: tezedge/tezedge-ci-builder:rust-1.58.1-v13.0-octez
+    pull: if-not-exists
+    user: root
+    environment:
+      RUST_BACKTRACE: 1
+      SODIUM_USE_PKG_CONFIG: 1
+    commands:
+      # copy the tezos src file from the image to a shared volume
+      - mkdir -p python-test-logs
+      - mkdir -p tezos-src
+      - cp -R /home/appuser/tezos-src/tezos tezos-src/tezos
+      # prepare modified python files for tests (compatibility with tezedge node)
+      - cp drone-cache/python-tests/daemons/*.py tezos-src/tezos/tests_python/daemons/
+      - cp drone-cache/python-tests/launchers/sandbox.py tezos-src/tezos/tests_python/launchers/
+      - cp drone-cache/python-tests/tools/constants.py tezos-src/tezos/tests_python/tools/
+      - cp drone-cache/sandbox_files/sandbox-patch-context.json tezos-src/tezos/
+      - cp drone-cache/python-tests/tests_012/*.py tezos-src/tezos/tests_python/tests_012/
+      # copy binaries to the test
+      - cp drone-cache/build_files/light-node tezos-src/tezos/
+      - cp drone-cache/build_files/protocol-runner tezos-src/tezos/
+      - cp drone-cache/build_files/tezedge-baker tezos-src/tezos/
+      - cp drone-cache/build_files/ffi/libtezos.so tezos-src/tezos/
+      # copy zcash-params init files - see constants.py light-node params
+      - cp drone-cache/build_files/ffi/sapling-spend.params tezos-src/tezos/
+      - cp drone-cache/build_files/ffi/sapling-output.params tezos-src/tezos/
+
+  - name: baker-endorser-test
+    <<: *python-test
+    environment:
+      PYTHON_TEST: test_baker_endorser.py
```

Comment:
Two questions:

- Does this really need a full dedicated pipeline instead of being added as an extra test to the already existing pipelines?
- Other than the environment differences (using our own baker instead of octez), is there any difference in the test code itself?

Reply:
I improved one test `test_baker_endorser.py`. And maybe improve more after this merged.